hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/authority/PKIProfiles/DefaultCAProfile.java

```diff
@@ -43,4 +47,38 @@
       VALIDATE_NAME_CONSTRAINTS = (e, b) -> TRUE;
   static final BiFunction<Extension, PKIProfile, Boolean>
       VALIDATE_CRL_DISTRIBUTION_POINTS = (e, b) -> TRUE;
-}
+
+
+  private static boolean validateBasicExtensions(Extension ext,
+      PKIProfile pkiProfile) {
+    BasicConstraints constraints =
+        BasicConstraints.getInstance(ext.getParsedValue());
+    if(constraints.isCA()) {
```

Comment:
My idea here is DefaultCAProfile when used by CA Server it will issue CA Certificate.
And RootCA Server right now starts with DefaultCAProfile on fresh installed clusters, where as on upgraded cluster from non-HA rootCAServer will be started like before with DefaultProfile. The integration of DefaultCAServer and starting 2 CA will be taken care in further Jiras.

Reply:
Yes.